inst/include/population.h

```diff
@@ -39,7 +39,15 @@
 namespace smc {
 
     /// A stable calculation of the log sum of the weights, used in ESS calculations
-    double stableLogSumWeights(const arma::vec & logw);
+    /// This function performs a stable calculation of the log sum of the weights, which is useful for
+    /// normalising weights, calculating the effective sample size and estimating the normalising constant.
+    ///
+    /// \param logw The log weights of interest.
+    inline double stableLogSumWeights(const arma::vec & logw){
+        double dMaxWeight = arma::max(logw);
+        double sum = arma::sum(exp(logw - dMaxWeight));
+        return (dMaxWeight + log(sum));
+    }
```

Comment:
I have a vague aesthetic objection to non-templated code in header files, but that's because I'm a dinosaur. With inline functions I guess it doesn't matter anyway, so I don't have a problem with it if @eddelbuettel  is happy. I can see at least small advantages in keeping the library side of the code header only if it doesn't cause other problems.